### PR TITLE
Clear State before raising errors

### PR DESCRIPTION
**Pull request by ElijahSwiftIBM**
Add state clearing before raising SegmentError or SegmentTraitError.

Add unit test for this (Unit test for SegmentTraitError includes extract to simulate functionality of add when generateRequestOnly is not set)

### :bulb: Issue Reference

**Issue:**  #33 

### :computer: What does this address?

SegmentError and SegmentTraitError exit the code without clearing segment trait dictionaries. This leads to improper request dictionaries built on the next request.

### :pager: Implementation Details

added call to clear_state() function before throwing errors.

### :clipboard: Is there a test case?

Built new test cases, one for SegmentError and one for SegmentTraitError, both hanging off of USER Admin

Test cases work as designed. Without clear_state() both fail, with it they succeed.

### CODE REVIEW

File: tests/user/test_user_request_builder.py [hunk 1]
```diff
@@ -140,6 +140,30 @@ def test_user_admin_build_add_request_with_bad_segment_traits(self):
...
+            user_admin.add(
+                "squidwrd", TestUserConstants.TEST_ADD_USER_REQUEST_BAD_TRAITS
+            )
+        self.assertEqual(
+            exception.exception.message,
+            "Unable to build Security Request.\n\n"
+            + f"'{bad_trait}' is not a known segment-trait "
+            + f"combination for '{self.user_admin._profile_type}'.\n",
+        )
+        result = user_admin.extract("squidwrd", segments=["omvs"])
```
Comment (lcarcaramo) [inline tests/user/test_user_request_builder.py:158]: Is there a reason for doing extract and add here?

Comment (ElijahSwiftIBM) [reply tests/user/test_user_request_builder.py:158]: Yes, in GenerateRequestOnly Mode, Add does not do a preliminary extract, but outside of this mode it does to know if it needs to throw AddOperationError(s). This extract emulates that, as this is necessary to see the impact of the bug.

Comment (lcarcaramo) [reply tests/user/test_user_request_builder.py:158]: Can you add a comment?

File: tests/user/test_user_request_builder.py [hunk 2]
```diff
@@ -140,6 +140,33 @@ def test_user_admin_build_add_request_with_bad_segment_traits(self):
             + f"combination for '{self.user_admin._profile_type}'.\n",
         )
 
+# Since this test uses GenerateRequestsOnly, the "Add" after the AddOperationError is returned
+# does not begin with an "Extract" call. This is necessary to recreate the error, so an extra
+# extract call was added to simulate this behavior.
```
Comment (lcarcaramo) [inline tests/user/test_user_request_builder.py:145]: Indent comment.

**Review by lcarcaramo — APPROVED** [review]

**Pull request merged**